rotkehlchen/api/v1/resources.py

```diff
@@ -2162,11 +2171,11 @@ def get(self, asset: Asset) -> Response:
 
         return self.rest_api.get_asset_icon(asset, match_header)
 
-    @use_kwargs(upload_schema, location='json_and_view_args')
+    @use_kwargs(upload_schema, location='json_and_query')
     def put(self, asset: Asset, file: Path) -> Response:
         return self.rest_api.upload_asset_icon(asset=asset, filepath=file)
 
-    @use_kwargs(upload_schema, location='view_args_and_file')
+    @use_kwargs(upload_schema, location='query_and_file')
```

Comment:
For consistency and because with the get method we can't pass forms or use json from the frontend part. In one of the resources where this is used we have all the verbs in use and get can't be changed to post to solve this issue so I believe query args is the best here

Reply:
But isn't it `post` verb here?

Reply:
Okay, we just misunderstood each other, never mind